packages/scandipwa/src/component/RecentlyViewedWidget/RecentlyViewedWidget.container.js

```diff
@@ -50,13 +53,14 @@ export class RecentlyViewedWidgetContainer extends PureComponent {
 
     componentDidMount() {
         const {
-            shouldBeUpdated,
+            updateLoadStatus,
             updateRecentViewedProductsInfo,
             recentProducts,
             store
         } = this.props;
 
-        if (shouldBeUpdated && Object.entries(recentProducts).length !== 0) {
+        if (Object.entries(recentProducts).length !== 0) {
+            updateLoadStatus(true);
```

Comment:
Just to make it more clear, can we set loading status in RecentlyViewedProducts.dispatcher ? I guess it can be set in prepareRequest function.